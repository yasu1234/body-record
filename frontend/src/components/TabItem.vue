<script setup>
import { ref } from 'vue'

const props = defineProps({
    menuTitle: String,
    menuId: Number,
    currentId: Number
})

const emit = defineEmits(['menuTapped']);

function handleClick(menuId) {
    emit('menuTapped', menuId);
}
</script>

<template>
    <div>
        <button @click="handleClick(props.menuId)" class="tab-button" :class="[props.currentId === props.menuId ? 'active' : '', 'tab']">
            {{props.menuTitle}}
        </button>
        <div class="active-tab" v-if="props.currentId === props.menuId"></div>
    </div>
</template>

<style>
.tab {
  border-radius: 2px 2px 0 0;
  background: #fff;
  color: #867d74;
  line-height: 24px;
}
.tab-button {
    margin-left: 8px;
    margin-right: 8px;
}
.tab:hover {
  color: #000000;
}
.active {
    color: #867d74;
}
.active-tab {
    height: 4px;
    background: #ffa500;
}
</style>