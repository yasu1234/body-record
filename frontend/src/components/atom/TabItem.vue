<template>
  <div>
    <button
      @click="handleClick(props.menuId)"
      class="tab-button"
      :class="[props.currentId === props.menuId ? 'active' : '', 'tab']"
    >
      {{ props.menuTitle }}
    </button>
    <div class="active-tab" v-if="props.currentId === props.menuId"></div>
  </div>
</template>

<script setup>
const props = defineProps({
  menuTitle: String,
  menuId: Number,
  currentId: Number,
});

const emit = defineEmits(["menu-tapped"]);

function handleClick(menuId) {
  emit("menu-tapped", menuId);
}
</script>

<style scoped>
.tab {
  background: #fff;
  color: #867d74;
}
.tab-button {
  margin-left: 8px;
  margin-right: 8px;
  padding: 0.6em 1.2em;
}
.tab:hover {
  color: #000000;
}
.active {
  color: #000000;
}
.active-tab {
  height: 4px;
  background: #ffa500;
  margin-top: 3px;
}
button:hover {
  opacity: 1;
}
</style>
