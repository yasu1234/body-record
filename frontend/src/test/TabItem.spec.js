import { shallowMount } from '@vue/test-utils'
import TabItem from '@/components/atom/TabItem.vue'

jest.mock('vue-router', () => {
  const router = {
    push: jest.fn(),
    pathname: '/',
    route: '',
    asPath: '',
    query: {},
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
  };
  return {
    useRouter: jest.fn().mockReturnValue(router),
  };
});

describe('TabItem.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(TabItem, {
      global: {
        stubs: ["router-link", "router-view"]
      }
    })
  })

  it('SnapShotテスト(activeになる)', () => {
    wrapper = shallowMount(TabItem, {
        props: {
          menuTitle: 'Home',
          menuId: 1,
          currentId: 1
        }
    })
    expect(wrapper.element).toMatchSnapshot();
  })

  it('SnapShotテスト(activeではない)', () => {
    wrapper = shallowMount(TabItem, {
        props: {
          menuTitle: 'Home',
          menuId: 1,
          currentId: 2
        }
    })
    expect(wrapper.element).toMatchSnapshot();
  })

  it('menuIdとcurrentIdが一致していたらactiveが適用される', async () => {
    wrapper = shallowMount(TabItem, {
      props: {
        menuTitle: 'Home',
        menuId: 1,
        currentId: 1
      }
    })
    const button = wrapper.find('button')
    expect(button.classes()).toContain('active')
  })
 
  it('menuIdとcurrentIdが一致していない場合にはactiveが適用されない', async () => {
    wrapper = shallowMount(TabItem, {
      props: {
        menuTitle: 'Home',
        menuId: 1,
        currentId: 2
      }
    })
    const button = wrapper.find('button')
    expect(button.classes()).not.toContain('active')
  })
 
  it('メニューボタンをタップしたときにmenuIdがemitで渡されていること', async () => {
    wrapper = shallowMount(TabItem, {
      props: {
        menuTitle: 'Home',
        menuId: 1,
        currentId: 1
      }
    })
    const button = wrapper.find('button')
    await button.trigger('click')
    expect(wrapper.emitted('menuTapped')).toBeTruthy()
    expect(wrapper.emitted('menuTapped')[0]).toEqual([1])
  })
})

