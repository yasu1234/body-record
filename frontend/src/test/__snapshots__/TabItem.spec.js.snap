// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabItem.vue SnapShotテスト(activeではない) 1`] = `
<div>
  <button
    class="tab-button tab"
  >
    Home
  </button>
  <!--v-if-->
</div>
`;

exports[`TabItem.vue SnapShotテスト(activeになる) 1`] = `
<div>
  <button
    class="tab-button active tab"
  >
    Home
  </button>
  <div
    class="active-tab"
  />
</div>
`;
